src/librustc/ty/mod.rs

```diff
@@ -2746,12 +2746,24 @@ fn associated_item_def_ids<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
     Rc::new(vec)
 }
 
+fn describe_def<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
+                          def_id: DefId)
+                          -> Option<Def> {
+   let id = tcx.hir.as_local_node_id(def_id).unwrap();
+   let item = tcx.hir.expect_item(id);
+   match item.node{
+       hir::ItemTrait(..) => Some(Def::Trait(def_id)),
+       _ => None
```

Comment:
@eddyb I hope this is the implementation you had it mind. It's the only way I could see of turning the `hir::map::Node` into a `Def`.

I also wasn't sure what would be the appropriate `Def` type to use for an `hir::ItemImpl` therefore I left it as none, but I would be happy to change it to any type you would suggest.